server/src/services/person.service.ts

```diff
@@ -88,7 +88,7 @@ export class PersonService {
   }
 
   async getAll(auth: AuthDto, dto: PersonSearchDto): Promise<PeopleResponseDto> {
-    const { machineLearning } = await this.configCore.getConfig();
+    const { machineLearning } = await this.configCore.getConfig({ withCache: false });
```

Comment:
I see that you're disabling caching for places where it won't get called repeatedly, but I don't see the harm of caching in places like here, in the init methods or when queueing.

Reply:
I guess it is a matter of opinion. There is a trade off between doing an extra database request sometimes vs the complexity and potential inconsistency that comes with caching the config. So far, there have already been several issues related to the fact that the config is cached. IMO it is clearer and simpler to just not cache it in general. I understand that there is some benefit to caching it in jobs that are called _a lot_, but outside of that I think it would be less error-prone to not worry about reading a cached value.

For initialization, we could load it once and pass it to each init method. That would be more clear/transparent in my mind.

Reply:
I guess I just think of it as when would you *not* want caching, and cache everything outside of that. There's nothing to gain from not caching in the queueing job when all the jobs it queues will use the cache. Passing the config to the init methods is fine, but it's just more code for the same effect.

Reply:
If caching always worked and always loaded the correct value I would agree with you, but it doesn't and it doesn't seem worthwhile to risk running into really weird issues because of it for the sake of saving a single database query from being made.

Reply:
The config core treats caching from the DB and the config file the same since there's no need to differentiate them right now. The more places you don't cache, the more redundant disk IO unless you handle those differently.

I'm okay with not caching for auth, administrative functions or notifications, but things that add latency to common user-facing calls should be cached unless there's a specific reason not to, including getAll, searchSmart, getTheme , getMapStyle, getCustomCss and the init methods.

Reply:
What type of latency are we talking about? Also, it should not write to WAL for doing a select should it?

Reply:
About a millisecond on a high-end machine with an NVME drive, probably twice that on spinning disks, a bit more than that if the config file is on a NAS or something, and an order of magnitude more if Postgres is remote.

> Also, it should not write to WAL for doing a select should it?

It writes to WAL for *everything*.

Reply:
This basically boils down to either "whitelisting" where we want caching or "blacklisting" where we don't want it. Obviously whitelisting is always less prone to error, since you cannot accidentally miss one spot. However, this also implies that the issue of caching in the wrong place is worse than the performance impact. Intuitively I would say that's the case, but I'm not sure.